va/va_enc_av1.h

```diff
@@ -426,19 +428,19 @@ typedef struct _VAEncSegParamAV1 {
  * #VAEncMacroblockMapBufferType and the driver as well as the underline encoder
  * should honor what is given by the app.
  */
-typedef struct _VAEncSegMapBufferAV1 {
-    /** \brief Segment map data size. */
-    uint32_t    segmentMapDataSize;
 
-    /**
-     * \brief Segment map.
-     * Size of this map is indicated by \ref segmentMapDataSize and each element
-     * in this map contains the segment id of a particular block.
-     * The element is indexed by raster scan order.
-     * The value of each entry should be in the range [0..7], inclusive.
-     */
-    uint8_t    *pSegmentMap;
-} VAEncSegMapBufferAV1;
+/**
+ * \brief AV1 Block Segmentation ID Buffer
+ *
+ * The application provides a buffer of VAEncMacroblockMapBufferType containing
```

Comment:
Do we need to set size and num_elements in vaCreateBuffer to width and height as well as in vp9, respectively?

Reply:
it is a kind of backward combability issue, I want to unify them to a linear buffer. so, the question is: is there other user beside intel media driver , is there any application is using the older interface?

Reply:
Would it be a better solution if we keep the original behavior when calling vaCreateBuffer with VAEncMacroblockMapBufferType, and we define a new BufferType (e.g. named VAEncUnifiedSegmentationMapBufferType). 
There are some benefits but libva will a little bit complicated.
We can keep all the backward compability and suggest/guide app-developers to switch to the new/unified interface.

Reply:
so, we support both for VP9?

Reply:
Yes, I would prefer that. 
It would benefit the application maintenance effort. 
However, it will make libva more complicated......